src/js/actions/tools.js

```diff
@@ -519,6 +519,7 @@ define(function (require, exports) {
      */
     var _toolModalStateChangedHandler,
         _vectorMaskHandler,
+        _vectorSelectMaskHandler,
```

Comment:
this is an old problem, obviously, but I wonder if these shouldn't get nulled out by onReset?